olive/systems/azureml/aml_system.py

```diff
@@ -612,10 +612,15 @@ def evaluate_pipeline():
             outputs = {}
             for metric in evaluator_config.metrics:
                 metric_tmp_dir = tmp_dir / metric.name
+                metric_clone = deepcopy(metric)
+                if metric_clone.sub_types:
+                    metric_clone.sub_types = sorted(metric_clone.sub_types, key=lambda st: st.priority, reverse=True)
+                    for priority, sub_type in enumerate(metric_clone.sub_types):
+                        sub_type.priority = priority + 1
```

Comment:
this works too or we can just assign all priorities to `-1` since the priority is irrelevant for the evaluation.

Reply:
I tried that and it doesn't work. The logic triggers this check -

https://github.com/microsoft/Olive/blob/b1e051e06122d914ca7aeaeabcc78486f476bb20/olive/evaluator/olive_evaluator.py#L1140

For it to work, I will have to remove the `name` field of the subtype.

Reply:
Oh that makes sense. Thanks! Approved